hera_cal/redcal.py

```diff
@@ -789,3 +837,335 @@ def is_redundantly_calibratable(antpos, bl_error_tol=1.0):
+    rv['chisq_per_ant'] = {ant: np.zeros((nTimes, nFreqs), dtype=np.float32) for ant in ants}
+
+    # get reds and then intitialize omnical visibility solutions to all 1s and all flagged
+    all_reds = get_reds({ant: hd.antpos[ant] for ant in ant_nums}, pol_mode=pol_mode,
+                        pols=set([pol for pols in pol_load_list for pol in pols]))
+    all_reds = filter_reds(all_reds, ex_ants=ex_ants)
+    rv['v_omnical'] = {red[0]: np.ones((nTimes, nFreqs), dtype=np.complex64) for red in all_reds}
+    rv['vf_omnical'] = {red[0]: np.ones((nTimes, nFreqs), dtype=bool) for red in all_reds}
+
+    # solar flagging
+    lat, lon, alt = hd.telescope_location_lat_lon_alt_degrees
+    solar_alts = utils.get_sun_alt(hd.times, latitude=lat, longitude=lon)
+    solar_flagged = solar_alts > solar_horizon
+    if verbose and np.any(solar_flagged):
+        print(len(hd.times[solar_flagged]), 'integrations flagged due to sun above', solar_horizon, 'degrees.')
+    
+    # loop over data, performing partial loading 
+    for pols in pol_load_list:
+        if verbose:
+            print('Now calibrating', pols, 'polarization(s)...')
+        reds = filter_reds(all_reds, ex_ants=ex_ants, pols=pols)
+        for tinds in np.split(np.arange(nTimes)[~solar_flagged], 
+                              np.arange(nInt_to_load, len(hd.times[~solar_flagged]), nInt_to_load)):
+            if len(tinds) > 0:
+                if verbose:
+                    print('    Now calibrating times', hd.times[tinds[0]], 'through', hd.times[tinds[-1]], '...')
+                data, _, _ = hd.read(times=hd.times[tinds], polarizations=pols)
+                cal = redundantly_calibrate(data, reds, freqs=hd.freqs, times_by_bl=hd.times_by_bl, 
+                                            conv_crit=conv_crit, maxiter=maxiter, 
+                                            check_every=check_every, check_after=check_after, gain=gain)
+                # gather results 
+                for ant in cal['g_omnical'].keys():
+                    rv['g_firstcal'][ant][tinds, :] = cal['g_firstcal'][ant]
+                    rv['gf_firstcal'][ant][tinds, :] = cal['g_firstcal'][ant]
+                    rv['g_omnical'][ant][tinds, :] = cal['g_omnical'][ant]
+                    rv['gf_omnical'][ant][tinds, :] = cal['gf_omnical'][ant]
+                    rv['chisq_per_ant'][ant][tinds, :] = cal['chisq_per_ant'][ant]
+                for bl in cal['v_omnical'].keys():
+                    rv['v_omnical'][bl][tinds, :] = cal['v_omnical'][bl]
+                    rv['vf_omnical'][bl][tinds, :] = cal['vf_omnical'][bl]
+                if pol_mode in ['1pol', '2pol']:
+                    for antpol in cal['chisq'].keys():
+                        rv['chisq'][antpol][tinds, :] = cal['chisq'][antpol]
+                else:  # duplicate chi^2 into both antenna polarizations
+                    for antpol in rv['chisq'].keys():
+                        rv['chisq'][antpol][tinds, :] = cal['chisq']
```

Comment:
Have we aggregated all of the metadata we want out of cal?  What about iterations, etc.?  Might help debugging or knowing if you're fully converged.

Reply:
It's a bit tricky. We'd need more code for building containers to hold other metadata and there isn't an obvious way to store it in calfits. I think we're OK for now, since this reproduces the omnical functionality we currently have, but maybe we should make this an issue?

Reply:
Moved to Issue #381.